sound/soc/codecs/hdac_hda.c

```diff
@@ -423,6 +431,16 @@ static int hdac_hda_codec_probe(struct snd_soc_component *component)
 		dev_err(&hdev->dev, "failed to create hda codec %d\n", ret);
 		goto error_no_pm;
 	}
+
+#ifdef CONFIG_SND_HDA_PATCH_LOADER
+	if (hda_pvt->fw) {
+		ret = snd_hda_load_patch(hcodec->bus, hda_pvt->fw->size, hda_pvt->fw->data);
+		if (ret < 0) {
+			dev_err(&hdev->dev, "failed to load hda patch %d\n", ret);
+			goto error_no_pm;
```

Comment:
Do you prefer to abort the normal probe with patching failure ?

Reply:
> Do you prefer to abort the normal probe with patching failure ?

Yes, this refers to the legacy HDA driver [if (err < 0) goto out_free;](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/pci/hda/hda_intel.c#L2317)